frontend/src/libs/components/prompt-generation/prompt-generation.tsx

```diff
@@ -0,0 +1,69 @@
+import { Button, Icon } from '~/libs/components/components.js';
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import { useCallback, useState } from '~/libs/hooks/hooks.js';
+
+import { PromptCard } from './libs/components/components.js';
+import { PromptCategory } from './libs/enums/enums.js';
+import { categoryPrompts } from './libs/mocks/mocks.js';
+import styles from './styles.module.scss';
+
+const { CHARACTER, SITUATION, SETTING, PROP, GENRE } = PromptCategory;
+
+const defaultPrompt = {
+  [CHARACTER]: 'Girl',
+  [SITUATION]: 'Street',
+  [SETTING]: '',
+  [PROP]: 'Red hair',
+  [GENRE]: 'Fiction',
+};
+
+const PromptGeneration: React.FC = () => {
+  const [isSpinning, setIsSpinning] = useState(false);
+  const [generatedPrompt, setGeneratedPrompt] = useState(defaultPrompt);
+
+  const promptCategories = Object.values(PromptCategory);
+
+  const handleButtonClick = useCallback((): void => {
+    setIsSpinning(true);
+
+    setTimeout(() => {
```

Comment:
It's a good practice to clear all timers in the `useEffect` cleanup function. Something like that
```js
useEffect(() => {
    const timerId = setTimeout(() => {}, 3000)
    
    return () => {
      clearTimeout(timerId)
    }
})
```

Reply:
Got it, thank you)

Reply:
It was an example, I should have mentioned that, sorry😅 In your case it should be like
```js
useEffect(() => {
  const timerId = setTimeout(() => {
    setGeneratedPrompt((previousGeneratedPrompt) => {
      const updatedGeneratedPrompt = { ...previousGeneratedPrompt };
      for (const category of promptCategories) {
        const prompts = categoryPrompts[category];
        const randomIndex = Math.floor(Math.random() * prompts.length);
        updatedGeneratedPrompt[category] = prompts[randomIndex];
      }
      return updatedGeneratedPrompt;
    });
    setIsSpinning(false);
  }, 3000)
  
  return () => {
     clearTimeout(timerId)
  }
}, [promptCategories])
```